### DISCUSSION TITLE
React Native focus next TextInput on Keyboard Next button press

### DISCUSSION THREAD

**Discussion by KiranRadish** [react-hook-form/react-hook-form#7818] [Q&A] [answered]
I have installed version 7 and trying to use ref but its not working (using javascript code not typescript).

I want to focus TextInput like we do in hooks with useRef

**Comment by bluebill1049** [comment]
can you share a codesandbox or snack?

**Reply by KiranRadish** [reply]
This is how my code is 

<Controller
            name="email"
            defaultValue=""
            control={control}
            rules={{
              required: { value: true, message: errorMessage.required },
              pattern: { value: Keys.EMAIL_REGEX, message: 'Not a valid email' }
            }}
            render={({ field: { onChange, onBlur, value, ref } }) => (
              <View style={[Metrics.mt6, Metrics.mb15]}>
                <AppTextInput
                  inputStyle={AppStyles.inputBox}
                  error={errors.email}
                  errorMessage={errors?.email?.message}
                  keyboardType={'email-address'}
                  onChangeText={(text) => onChange(text)}
                  value={value}
                  onBlur={onBlur}
                  returnKeyLabel="next"
                  returnKeyType="next"
                  autoCapitalize={'none'}
                  // ref={register({ name: 'email' })}
                />
                
                
                <Controller
            name="password"
            defaultValue=""
            control={control}
            rules={{ required: { value: true, message: errorMessage.required }, }}
            render={({ field: { onChange, onBlur, value, ref } }) => (
              <View style={[Metrics.mt6, Metrics.mb15]}>
                <AppTextInput
                  inputStyle={AppStyles.inputBox}
                  error={errors.password}
                  errorMessage={errors?.password?.message}
                  keyboardType={'default'}
                  secureTextEntry={true}
                  onChangeText={(text) => onChange(text)}
                  value={value}
                  onBlur={onBlur}
                  returnKeyLabel="next"
                  returnKeyType="next"
                  autoCapitalize={'none'}
                // ref={ref}
                />
              </View>

            )}
          />

I want to focus password textInput on press of Next button when keyboard opens 
we will use onSubmitEditing for that but here how can i implement its throwing error

**Comment by Moshyfawn** [comment]
You gotta show us some usage code _([codesanbox](https://codesandbox.io/s/react-hook-form-v7-js-template-quqov) ideally)_. We can't magically know what your issue is ;)

In V7, for uncontrolled components, instead of `ref={register}` you need to do `<input {...register("name")} />`. For controlled, it's `render={({ field: { ref } }) => }` instead of `render={{ ref }}`

If your input exposes a different ref prop name instead of `ref`, you can do
```jsx
const { nameFieldRef, ...nameField } = register("name")

<input { ...nameField } inputRef={nameFieldRef} />
```

**Comment by KiranRadish** [comment, marked as answer]
I got the answer, Thanks man for the wonderful library

**Reply by vagnerwentz** [reply]
Hi, could you share the answer for your question?

**Reply by pke** [reply]
@KiranRadish so what's the answer/solution to your problem?

**Reply by KiranRadish** [reply]
@vagnerwentz & @pke  the answer worked for me is -


we have to use `setFocus` from useForm.

In render we can use it like this

`**render={({ field: { onChange, onBlur, value, ref } }) => (**

             <View style={[Metrics.mt3, Metrics.mb15]}>
                      <AppTextInput
                        inputStyle={AppStyles.inputBox}
                        error={errors.firstName}
                        errorMessage={errors?.firstName?.message}
                        keyboardType={'default'}
                        onChangeText={text => onChange(text)}
                        value={value}
                        onBlur={onBlur}
                        returnKeyLabel="next"
                        returnKeyType="next"
                        autoCapitalize={'none'}
                        ref={ref}
                        onSubmitEditing={() => setFocus('lastName')} // next defaultValues key for focus
                      />
              </View>
          )}
          
`


**onSubmitEditing={() => setFocus('lastName')}**

**Reply by praweshlamsal** [reply]
Is there any global way of achieving this, as i have lots of text input in my application and mannually setting setfocus is pretty tedious?
